src/test/java/core/basesyntax/service/TransactionProcessorTest.java

```diff
@@ -0,0 +1,111 @@
+package core.basesyntax.service;
+
+import static org.junit.Assert.assertTrue;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.impl.FileReaderServiceImpl;
+import core.basesyntax.service.impl.TransactionProcessorImpl;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.impl.BalanceTransactionImpl;
+import core.basesyntax.strategy.impl.PurchaseTransactionImpl;
+import core.basesyntax.strategy.impl.ReturnTransactionImpl;
+import core.basesyntax.strategy.impl.SupplyTransactionImpl;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.Before;
+import org.junit.Test;
+
+public class TransactionProcessorTest {
+    private static final String testTransactionFilePath = "src/main/resources/transaction.csv";
+    private static final int FRUIT_OPERATION_INDEX = 0;
+    private static final String TEST_FILE_HEADER = "fruit,quantity";
+    private final Map<String, OperationHandler> operationHandlerMap = new HashMap<>();
+
+    @Before
+    public void setUp() {
+        operationHandlerMap.put(Operation.BALANCE.getOperation(), new BalanceTransactionImpl());
+        operationHandlerMap.put(Operation.PURCHASE.getOperation(), new PurchaseTransactionImpl());
+        operationHandlerMap.put(Operation.SUPPLY.getOperation(), new SupplyTransactionImpl());
+        operationHandlerMap.put(Operation.RETURN.getOperation(), new ReturnTransactionImpl());
+    }
+
+    @Test
+    public void allOperationElementsInListValid_OK() {
+        boolean isValid = true;
+        FileReaderService fileReaderService = new FileReaderServiceImpl();
+        List<String> testListOfTransaction = fileReaderService
+                .readFromFile(testTransactionFilePath);
+        for (int i = 1; i < testListOfTransaction.size(); i++) {
+            String[] testArrayFromList = testListOfTransaction.get(i).split(",");
+            if (!operationHandlerMap
+                    .containsKey(testArrayFromList[FRUIT_OPERATION_INDEX])) {
+                isValid = false;
+            }
+        }
+        assertTrue(isValid);
+    }
```

Comment:
it's a redundant test, you created test method for each of operations below already

Reply:
Done